src/ducks/Studio/index.js

```diff
@@ -40,44 +40,46 @@ const Studio = ({
   const [activeMetrics, setActiveMetrics] = useState(defaultMetrics)
   const [activeEvents, setActiveEvents] = useState(defaultEvents)
   const [MetricSettingMap, setMetricSettingMap] = useState(
-    defaultMetricSettingsMap,
+    defaultMetricSettingsMap
   )
   const [chartSidepane, setChartSidepane] = useState()
   const [advancedView, setAdvancedView] = useState()
   const [shareLink, setShareLink] = useState()
   const [isICOPriceDisabled, setIsICOPriceDisabled] = useState(true)
-  const [isSidebarClosed, setIsSidebarClosed] = useState()
+
   const [rawData, loadings, ErrorMsg] = useTimeseries(
     activeMetrics,
     settings,
-    MetricSettingMap,
+    getPreparedMetricSettings(activeMetrics, MetricSettingMap)
```

Comment:
`getPreparedMetricSettings` will be called on every render, which is bad. It should be a hook, which depends on the `activeMetrics` and `MetricSettingMap`